agent/structs/config_entry.go

```diff
@@ -305,15 +305,11 @@ func DecodeConfigEntry(raw map[string]interface{}) (ConfigEntry, error) {
 		return nil, err
 	}
 
-	for _, k := range md.Unused {
-		switch {
-		case k == "CreateIndex" || k == "ModifyIndex":
-		case strings.HasSuffix(strings.ToLower(k), "namespace"):
-			err = multierror.Append(err, fmt.Errorf("invalid config key %q, namespaces are a consul enterprise feature", k))
-		default:
-			err = multierror.Append(err, fmt.Errorf("invalid config key %q", k))
-		}
+	multiErr := validateUnusedKeys(md.Unused)
+	if multiErr != nil {
+		err = multierror.Append(err, multiErr)
```

Comment:
I missed this when I first looked at it. I think `mutlierror` is no longer necessary here, because we will never get to this line if `err != nil`.

So I think this can be:

```go
if err := validateUnusedKeys(md.Unused); err != nil {
    return nil, err
}
```

Reply:
That's right, updated this.